buildSrc/src/main/kotlin/jni-library.gradle.kts

```diff
@@ -0,0 +1,86 @@
+import org.gradle.internal.jvm.Jvm
+
+plugins {
+    `cpp-library`
+    `java-library`
+}
+
+val jniImplementation by configurations.creating
+
+configurations.matching {
+    it.name.startsWith("cppCompile") ||
+            it.name.startsWith("nativeLink") ||
+            it.name.startsWith("nativeRuntime")
+}.all {
+    extendsFrom(jniImplementation)
+}
+
+val jniHeaderDirectory = layout.buildDirectory.dir("jniHeaders")
```

Comment:
It seems lime this isn't used anywhere. Is it needed?

Reply:
Removed